### PR TITLE
[CELEBORN-630] Binary release artifact should package all versions of Spark and Flink clients

### PR DESCRIPTION
**Pull request by zhongqiangczq**
…link version

### CODE REVIEW

File: build/make-distribution.sh [hunk 1]
```diff
@@ -215,8 +215,9 @@ function build_flink_client {
 if [ "$RELEASE" == "true" ]; then
   build_service
   build_spark_client -Pspark-2.4
-  build_spark_client -Pspark-3.3
-  build_flink_client -Pflink-1.14
+  build_spark_client -Pspark-3.4
+  build_flink_client -Pflink-1.15
```
Comment (pan3793) [inline build/make-distribution.sh:219]: why not include flink 1.14?

Comment (zhongqiangczq) [reply build/make-distribution.sh:220]: yes, it should include. 
At before, i think flink-1.15 will be backward compatible flink-1.14, so flink-1.14 can be removed. I check again that flink-1.14 is little not compatible with flink-1.15. so flink-1.14 must be included.

**Review by pan3793 — APPROVED** [review]
Life will be easy if we can add e2e cross-version test and address all version-specific differences by introducing Shim interfaces or leveraging reflection.

**Review by pan3793 — APPROVED** [review]
Life will be easy if we can add e2e cross-version test and address all version-specific differences by introducing Shim interfaces or leveraging reflection.

**Comment by pan3793** [comment]
Thanks, merge to main/0.3

**Pull request closed without merging**